src/components/mols/NavBarLogged/index.tsx

```diff
@@ -25,8 +26,14 @@ const NavBarLogged: FC<NavBarLoggedProps> = ({ hasLogoImage = true }) => {
           />
         </Link>
       )}
-
-      <AvatarDropdown />
+      <div className="flex items-center gap-3">
+        {session?.user && (
+          <PremiumBadge
+            hasPremium={isDateValid(session?.user.subscriptionEndDate)}
```

Comment:
isDateValid = a data é valida?
Se for *qualquer data* esse função deveria retornar true pelo nome dela.

Sugestão é criar uma função: isPremiumUser(user)

Reply:
Ainda não temos na sessão um boolean  dizendo se é subscrito ou não, então a validação é feita se a data dessa subscrição é maior ou não que a data atual. Renomeei de uma forma adequada o isDateValid para hasDateExpired e criei o isPremiumUser